components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -73,65 +82,242 @@ public AbstractHTTPFunction() {
         allowedDomains = ConfigProvider.getInstance().getAllowedDomainsForHttpFunctions();
     }
 
-    protected void executeHttpMethod(HttpUriRequest request, Map<String, Object> eventHandlers) {
+    protected void executeHttpMethod(HttpUriRequest clientRequest, Map<String, Object> eventHandlers,
+                                     AuthConfigModel authConfigModel) {
 
         AsyncProcess asyncProcess = new AsyncProcess((context, asyncReturn) -> {
-            JSONObject json = null;
-            int responseCode;
             String outcome;
             String endpointURL = null;
 
-            if (request.getURI() != null) {
-                endpointURL = request.getURI().toString();
+            HttpUriRequest request;
+            try {
+                if (authConfigModel != null) {
+                    AuthConfig authConfig = AuthConfigFactory.getAuthConfig(authConfigModel, context, asyncReturn);
+                    request = authConfig.applyAuth(clientRequest, authConfigModel);
+                } else {
+                    request = clientRequest;
+                }
+
+                if (request.getURI() != null) {
+                    endpointURL = request.getURI().toString();
+                }
+
+                if (!isValidRequestDomain(request.getURI())) {
+                    LOG.error("Request URL does not match with the allowed domain list. Request Url: " +
+                            endpointURL);
+                    asyncReturn.accept(context, Collections.emptyMap(), Constants.OUTCOME_FAIL);
+                } else {
+                    Pair<Boolean, Pair<String, JSONObject>> result = executeRequest(request, endpointURL);
+                    if (Boolean.TRUE.equals(result.getLeft())) {
+                        LOG.error("Error while calling endpoint. Url: " + endpointURL);
```

Comment:
Rather than an error better log this as an info, indicating "Failed to invoke the endpoint. Url: {endpoint}. Retrying the request."

Reply:
Rather than just defining a boolean as "true" to decide on whether to retry or not, if an enum could be defined with a boolean as a param or as a string literal to denote whether to retry or not would improve the readability.

Reply:
Resolved with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/01b5e2f5f849d7a4e89d792e28845597544fca39